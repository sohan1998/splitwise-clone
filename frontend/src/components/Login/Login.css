.container {
    margin-top: 6rem;
    margin-left: 30rem;
}


.btn{
    margin-top: 0.7rem;
    background-color: #FF652F; 
    color: white
}

.inputLogin{
    font-size: 12pt;
    font-weight: 500
}