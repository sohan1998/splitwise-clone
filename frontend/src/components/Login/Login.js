import React, { Component } from "react";
import logo from "../assets/login_logo.png";
import "../Login/Login.css";
import axios from 'axios';
import cookie from 'react-cookies';
import {Redirect} from 'react-router';
import backendServer from "../../webConfig";
import PropTypes from 'prop-types';
import { connect } from "react-redux";
import {login} from "../../actions/actions.js";
var loginsuccess = ""


class Login extends Component {
  constructor(props) {
    super(props);
    // this.handleLogout = this.handleLogout.bind(this);
    this.state = {
      email: null,
      password: null,
      authFlag: false,
      //emailerror:"",
      //passsworderror:"", 
  }
    this.emailChangeHandler = this.emailChangeHandler.bind(this);
    this.passwordChangeHandler = this.passwordChangeHandler.bind(this);
    this.submitOwnerLogin = this.submitOwnerLogin.bind(this);
    
  }

emailChangeHandler = (e) => {
  this.setState({
      email : e.target.value
  })
}
passwordChangeHandler = (e) => {
  this.setState({
      password : e.target.value
  })
}

submitOwnerLogin = (e) => {
  var headers = new Headers();
  console.log("Inside submitowner")
  e.preventDefault();
  const data = {
      email : this.state.email,
      password : this.state.password
  }
  this.props.login(data)
}


componentWillReceiveProps(nextProps) {
  console.log("JSON.stringify(nextProps)", JSON.stringify(nextProps))
  if (nextProps.ld.logindetails === true) {
    loginsuccess = "true"
    console.log("loginsuccess", loginsuccess)
    console.log("nextProps.logindetails", nextProps.ld.logindetails)
    //this.props.posts.unshift(nextProps.newPost);
  }
  else
    loginsuccess = "false"
}


  render() {
    var redirectVar = null;
    
    if(localStorage.getItem('myjwttoken') && localStorage.getItem('useremail') ){
          console.log("Local storage found")
          redirectVar = <Redirect to= "/dashboard"/>
        }
    return (
     <div className="container">
      {redirectVar}
        <div className="row">
          <img src={logo} class="col-md-3" height="240" alt="Splitwise"></img>
          <div className="col-md-4">
            <h4 data-testid="LoginTest" style={{ color: "gray", fontSize: 19, marginBottom: 22 }}>
              WELCOME TO SPLITWISE
            </h4>
            <form>
              <div class="form-group inputLogin">
                <label for="exampleInputEmail1">Email address</label>
                <input
                  type="email"
                  class="form-control"
                  id="exampleInputEmail1"
                  aria-describedby="emailHelp"
                  placeholder="Enter email"
                 onChange = {this.emailChangeHandler}
                 //errorText={this.state.emailerror}
                />
              </div>
              <div class="form-group inputLogin">
                <label for="exampleInputPassword1">Password</label>
                <input
                  type="password"
                  class="form-control"
                  id="exampleInputPassword1"
                  placeholder="Password"
                  onChange = {this.passwordChangeHandler}
                 // errorText={this.state.passsworderror}
                />
              </div>
              <button onClick={this.submitOwnerLogin} class="btn">
                Log in
              </button>
              <p style={{marginTop: 10}}>
                Forgot your password? <a href="#">Click here</a>
              </p>
            </form>
          </div>
        </div>
      </div>
    );
  }
}

Login.propTypes = {
  login : PropTypes.func.isRequired,
  ld : PropTypes.string
}

const mapStateToProps = state => ({
ld : state.login
})

export default connect(mapStateToProps, {login})(Login); 