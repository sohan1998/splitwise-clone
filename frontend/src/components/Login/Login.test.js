/* eslint-disable react/prop-types */
/* eslint-disable no-undef */
import React from 'react';
import { render, cleanup } from '@testing-library/react';
import TestRenderer from 'react-test-renderer';
//import { createStore } from 'redux';
import { MemoryRouter } from 'react-router-dom';
// import configureStore from 'redux-mock-store';
//import { Provider } from 'react-redux';
import '@testing-library/jest-dom';
//import allReducers from '../../reducers';
import NavbarClass from '../Navbar/Navbar';
import Login from '../Login/Login';

//const store = createStore(allReducers);

const component = TestRenderer.create(
  //<Provider store={store}>
    <MemoryRouter>
      <Login>
     </Login>
    </MemoryRouter>
 // </Provider>
);

// eslint-disable-next-line no-undef
afterEach(cleanup);

it('renders', async () => {
  expect(component.toJSON()).toMatchSnapshot();
});

test('Check for header', async () => {
  const { getByTestId } = render(
  //  <Provider store={store}>
      <MemoryRouter>
        <Login>

        </Login>
      </MemoryRouter>
    //</Provider>
  );
  expect(getByTestId('LoginTest')).toHaveTextContent('WELCOME TO SPLITWISE');
});