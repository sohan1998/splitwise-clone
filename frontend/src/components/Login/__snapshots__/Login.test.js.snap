// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <img
      alt="Splitwise"
      class="col-md-3"
      height="240"
      src="login_logo.png"
    />
    <div
      className="col-md-4"
    >
      <h4
        data-testid="LoginTest"
        style={
          Object {
            "color": "gray",
            "fontSize": 19,
            "marginBottom": 22,
          }
        }
      >
        WELCOME TO SPLITWISE
      </h4>
      <form>
        <div
          class="form-group inputLogin"
        >
          <label
            for="exampleInputEmail1"
          >
            Email address
          </label>
          <input
            aria-describedby="emailHelp"
            class="form-control"
            id="exampleInputEmail1"
            onChange={[Function]}
            placeholder="Enter email"
            type="email"
          />
        </div>
        <div
          class="form-group inputLogin"
        >
          <label
            for="exampleInputPassword1"
          >
            Password
          </label>
          <input
            class="form-control"
            id="exampleInputPassword1"
            onChange={[Function]}
            placeholder="Password"
            type="password"
          />
        </div>
        <button
          class="btn"
          onClick={[Function]}
        >
          Log in
        </button>
        <p
          style={
            Object {
              "marginTop": 10,
            }
          }
        >
          Forgot your password? 
          <a
            href="#"
          >
            Click here
          </a>
        </p>
      </form>
    </div>
  </div>
</div>
`;